varnish-macros/src/stats.rs

```diff
@@ -0,0 +1,162 @@
+use crate::parser_utils::parse_doc_str;
+use serde::Serialize;
+use syn::{punctuated::Punctuated, token::Comma, Data, Field, Fields, Type};
+
+type FieldList = Punctuated<Field, Comma>;
+
+#[derive(Serialize, Clone)]
+struct VscMetricDef {
+    pub name: String,
+    #[serde(rename = "type")]
+    pub counter_type: String, // "counter", "gauge"
```

Comment:
`counter_type` should be an enum. Also, perhaps call it `metric_type` for readability?